pkg/apis/provisioning/v1alpha5/constraints.go

```diff
@@ -51,22 +51,42 @@ func (c *Constraints) ValidatePod(pod *v1.Pod) error {
 	if err := c.Taints.Tolerates(pod); err != nil {
 		return err
 	}
-	// Test if pod requirements are valid
 	requirements := NewPodRequirements(pod)
+	// Test if pod requirements are valid
 	if err := requirements.Validate(); err != nil {
 		return fmt.Errorf("invalid requirements, %w", err)
 	}
-	// Test if pod requirements are compatible
+	// Test if pod requirements are compatible to the provisioner
 	if errs := c.Requirements.Compatible(requirements); errs != nil {
 		return fmt.Errorf("incompatible requirements, %w", errs)
 	}
 	return nil
 }
 
+func (c *Constraints) GenerateLabels() map[string]string {
+	labels := map[string]string{}
+	for key, value := range c.Labels {
+		labels[key] = value
+	}
+	for key := range c.Requirements.Keys() {
+		if !IsRestrictedNodeLabel(key) {
+			values := c.Requirements.Get(key)
```

Comment:
can you inline values, since it's only used in one place.

Reply:
done